### PR TITLE
Add instrumentation for the pgx package

### PR DESCRIPTION
**Pull request by MrAlias**
- [x] README
- [x] End-to-end tests
- [x] Unit tests
- [x] Code complete
- [x] Documented

### CODE REVIEW

File: instrumentation/github.com/jackc/pgx/splunkpgx/sql_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				Port:             5432,
+				Transport:        transport.TCP,
+			},
+		},
+		{
+			name: "url: tcp address",
+			dsn:  "postrgres://user:password@localhost:8080/testdb",
+			connCfg: splunksql.ConnectionConfig{
+				Name:             "testdb",
+				ConnectionString: "postrgres://user@localhost:8080/testdb",
```
Comment (MrAlias) [inline instrumentation/github.com/jackc/pgx/splunkpgx/sql_test.go:54]:
```suggestion
			dsn:  "postgres://user:password@localhost:8080/testdb",
			connCfg: splunksql.ConnectionConfig{
				Name:             "testdb",
				ConnectionString: "postgres://user@localhost:8080/testdb",
```

File: instrumentation/github.com/jackc/pgx/splunkpgx/sql.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+//     // ...
+package splunkpgx
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/jackc/pgx"
+	// Make sure to import this so the instrumented driver is registered.
+	_ "github.com/jackc/pgx/v4"
```
Comment (pellared) [inline instrumentation/github.com/jackc/pgx/splunkpgx/sql.go:48]: I think it should be

```suggestion
	_ "github.com/jackc/pgx/v4/stdlib"
```

File: instrumentation/github.com/jackc/pgx/splunkpgx/sql.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+// package when using database/sql.
+//
+// To use this package, replace any blank identified imports of the
+// github.com/jackc/pgx package with an import of this package and
+// use the splunksql.Open function as a replacement for any sql.Open function
+// use. For example, if your code looks like this to start.
+//
+//     import (
+//     	"database/sql"
+//     	_ "github.com/jackc/pgx"
```
Comment (pellared) [inline instrumentation/github.com/jackc/pgx/splunkpgx/sql.go:25]: It should be

```suggestion
	_ "github.com/jackc/pgx/v4/stdlib"
```

or 

```suggestion
	_ "github.com/jackc/pgx/stdlib"
```

before Go Modules were introduced (should we even handle such old versions as https://github.com/jackc/pgx/releases/tag/v3.5.0 ?)

Comment (MrAlias) [reply instrumentation/github.com/jackc/pgx/splunkpgx/sql.go:25]: This is not what the [README](https://github.com/jackc/pgx#example-usage) or [wiki](https://github.com/jackc/pgx/wiki/Getting-started-with-pgx#hello-world-from-postgresql) include. Do you have an example or docs showing this use of `stdlib`?

Comment (pellared) [reply instrumentation/github.com/jackc/pgx/splunkpgx/sql.go:25]: The README and wiki shows non-`database/sql"` API. You can test it 😉

Comment (pellared) [reply instrumentation/github.com/jackc/pgx/splunkpgx/sql.go:25]: https://github.com/jackc/pgx/blob/master/stdlib/sql.go

**Review by pellared — APPROVED** [review]

**Pull request merged**